version: '3.9'

services:
  userapipostgres:
    image: 'postgres:14-alpine'
    ports:
      - "5439:5439"
    container_name: userapipostgres
    environment:
      - POSTGRES_DB=userapipostgres
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=root
      - PGPORT=5439
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - user-api
    volumes:
      - db-data:/var/lib/postgresql/data

  pgadmin:
    container_name: pgadmin4_container
    image: dpage/pgadmin4
    restart: on-failure
    ports:
      - "5050:80"
    depends_on:
      - userapipostgres
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: root
    networks:
      - user-api
    volumes:
      - pgadmin-data:/var/lib/pgadmin

  userapi:
    image: 'docker-spring-boot-postgres:latest'
    build:
      context: .
      dockerfile: userapi.Dockerfile
    container_name: userapi
    ports:
      - "8080:8080"
    depends_on:
      userapipostgres:
        condition: service_healthy
    networks:
      - user-api

networks:
  user-api: {}

volumes:
  db-data:
    external: false
  pgadmin-data:
    external: false
